lapwing-prepare-optimized-svg-for-react-component.rb

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env ruby
+
+# require 'pry'
+require 'fileutils'
+require 'nokogiri'
+
+if ARGV.size < 2
+  $stderr.puts "Usage: ruby ./lapwing-prepare-optimized-svg-for-react-component.rb source-svgs/lapwing.svg target-js/LapwingStenoDiagram.js"
+  exit 1
+end
+
+SOURCE_SVG = ARGV[0]
+TARGET_JS = ARGV[1]
+
+# #STKPWHRAO*-EUFRPBLGTSDZ
+KEYS = ["numberBarKey", "leftSKey", "leftTKey", "leftKKey", "leftPKey", "leftWKey", "leftHKey", "leftRKey", "leftAKey", "leftOKey", "starKey", "rightEKey", "rightUKey", "rightFKey", "rightRKey", "rightPKey", "rightBKey", "rightLKey", "rightGKey", "rightTKey", "rightSKey", "rightDKey", "rightZKey"]
+
+SYMBOLS = ["numberBar", "leftS", "leftT", "leftK", "leftP", "leftW", "leftH", "leftR", "leftA", "leftO", "star", "rightE", "rightU", "rightF", "rightR", "rightP", "rightB", "rightL", "rightG", "rightT", "rightS", "rightD", "rightZ"]
+
+color_config = {}
+
+KEYS.each do | key |
+  color_config["#{key}OnColor"] = "#7109AA"
+  color_config["#{key}OffColor"] = "#E9D9F2"
+end
+
+SYMBOLS.each do | key |
+  color_config["#{key}OnColor"] = "#FFFFFF"
+  color_config["#{key}OffColor"] = "#FFFFFF"
+end
+
+# This is mostly irrelevant in the front end now
+SVG_WIDTH = 140
+
+source_svg_basename = File.basename(SOURCE_SVG)
+OPTIMIZED_SVG = "./optimized-svgs/#{source_svg_basename}"
+
+if !system "node_modules/.bin/svgo --pretty --config=svgo.config.mjs -i #{SOURCE_SVG} -o #{OPTIMIZED_SVG} > /dev/null"
+  exit 1
+end
+
+@doc = File.open(OPTIMIZED_SVG) { |f| Nokogiri::XML(f) }
+
+svg = @doc.at_css "svg"
+svg["width"] = SVG_WIDTH
+
+title = @doc.at_css "title"
+# title_content = title.content
+title.remove unless title == nil
+
+g = @doc.at_css "g"
+# g_id = g["id"]
+# Use this to offset Danish diagram and others by 1 pixel
+# g["transform"] = "translate(1 1)"
+g["id"] = "xxxstenoboard-xxx + this.props.brief xxx}"
+# end
+
+vars = {}
+
+# STENO KEYS
+rects = @doc.css "rect"
+rects.each do | rect |
+  rect_id = rect["id"]
+
+  # steno key strokes
+  stroke = rect["stroke"]
+  stroke_var_name = rect_id + "StrokeColor"
+  stroke_var_value = stroke
+  rect["stroke"] = "xxx{" + stroke_var_name + "xxx}"
+  vars.store(stroke_var_name, stroke_var_value)
+
+  # steno key fills
+  # key_fill = rect["fill"]
+  key_fill_var_name_on = rect_id + "OnColor"
+  key_fill_var_name_off = rect_id + "OffColor"
+  # key_fill_var_value = key_fill
+  rect["fill"] = "xxx{this.props." + rect_id + " ? " + key_fill_var_name_on + " : " + key_fill_var_name_off + "xxx}"
+  vars.store(key_fill_var_name_on, color_config[key_fill_var_name_on])
+  vars.store(key_fill_var_name_off, color_config[key_fill_var_name_off])
+end
+
+
+
+# STENO LETTERS
+paths = @doc.css "path"
+paths.each do | path |
+  path_id = path["id"]
+
```

Comment:
We can add this line here to automatically remove the unnecessary style attributes from `path` elements (and repeat this for any other elements that need `style` removed):

```suggestion

path.remove_attribute("style")
```

Reply:
I have added this for now.  Since this appears to be an artefact of using Inkscape, and I might give up on the Inkscape idea after all, we could possibly remove it later.  Will keep in mind during testing and revisit.

Reply:
ended up adding this to 'rect' as well as the 'path' section of the code.